### PR TITLE
Introduce prettier as JS linter

### PR DESCRIPTION
**Pull request by tnir**
## Contributor problem

JavaScript files are not so consistent.

### What was your diagnosis of the problem?

Any linter for JavaScript files is not setup yet.

### What is your fix for the problem, implemented in this PR?

Introduce prettier as a linter for JavaScript files.

### Why did you choose this fix out of the possible options?

`singleQuote` looks preferred unlike [Prettier default](https://prettier.io/docs/en/options.html#quotes), so `singleQuote` is set `true` in this PR at this moment.

Signed-off-by: Takuya Noguchi [takninnovationresearch@gmail.com](https://github.com/sponsors/tnir)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -32,3 +32,15 @@ jobs:
...
           NO_CONTRACTS: "true"
+
+  prettier:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: Install prettier 2.7.1
```
Comment (deivid-rodriguez) [inline .github/workflows/build.yml:43]: I guess "Install prettier" so you only need to update when place when you upgrade it?

**Review by deivid-rodriguez — COMMENTED** [review]
Sounds good!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -32,3 +32,15 @@ jobs:
...
+        run: npm install prettier@2.7.1
```
Comment (simi) [inline .github/workflows/build.yml:44]: Any reason to not make it part of devDeps in `package.json` actually and track version in there?

Comment (deivid-rodriguez) [reply .github/workflows/build.yml:44]: Good idea, indeed, and will probably work better if we want dependabot to send us updates in the future!

File: .github/workflows/build.yml (hunk 1 above)
Comment (tnir) [reply .github/workflows/build.yml]: Makes sense.

File: .github/workflows/build.yml (hunk 2 above)
Comment (tnir) [reply .github/workflows/build.yml]: Makes sense.

**Comment by tnir** [comment]
@deivid-rodriguez @simi Thank you for your feedback.

**Comment by simi** [comment]
> @deivid-rodriguez @simi Thank you for your feedback.

It was a pleasure. Btw. we can add this commit to `.git-blame-ignore-revs` as well. But that is totally optional in here.

**Comment by tnir** [comment]
I do not think it should do so as changes in `assets/javascripts/*.js` in b94c55b197e3be0416cfb9aefc44a31b64d58f4c is only in 81 lines (and .github/workflows was slightly improved).
```
$ git show b94c55b197e3be0416cfb9aefc44a31b64d58f4c --numstat
commit b94c55b197e3be0416cfb9aefc44a31b64d58f4c
Author: Takuya Noguchi <takninnovationresearch@gmail.com>
Date:   Sat Jul 23 15:13:01 2022 +0000

    Add prettier as JS linter
    
    Signed-off-by: Takuya Noguchi <takninnovationresearch@gmail.com>

12      0       .github/workflows/build.yml
1       0       .prettierrc
3       3       assets/javascripts/application.js
4       2       assets/javascripts/commands_layout.js
39      33      assets/javascripts/search.js
35      26      assets/javascripts/search_arrows.js
22      0       package-lock.json
4       2       package.json
```
Thanks.

**Comment by deivid-rodriguez** [comment]
Yeah, I think using `.git-blame-ignore-revs` should only be used for huge commits.

**Pull request merged**